cmake/modules/SIMDExt.cmake

```diff
@@ -15,34 +15,30 @@
 #
 
 if(CMAKE_SYSTEM_PROCESSOR MATCHES "aarch64|AARCH64")
-  set(HAVE_ARM 1) 
-  set(save_quiet ${CMAKE_REQUIRED_QUIET})
-  set(CMAKE_REQUIRED_QUIET true)
-  include(CheckCXXSourceCompiles)
-  check_cxx_source_compiles("
-    #define CRC32CX(crc, value) __asm__(\"crc32cx %w[c], %w[c], %x[v]\":[c]\"+r\"(crc):[v]\"r\"(value))
-    asm(\".arch_extension crc\");
-    unsigned int foo(unsigned int ret) {
-      CRC32CX(ret, 0);
-      return ret;
-    }
-    int main() { foo(0); }" HAVE_ARMV8_CRC)
-  set(CMAKE_REQUIRED_QUIET ${save_quiet})
+  set(HAVE_ARM 1)
+  CHECK_C_COMPILER_FLAG(-march=armv8-a+crc HAVE_ARMV8_CRC)
```

Comment:
You are right! we need to consider the compatibility with old gcc. I have refined the patch so that it can switch to original implementation if old gcc doesn't recognize compile flags of "+crc+crypto".